docs/labs/handling-errors.html

```diff
@@ -0,0 +1,206 @@
+  examples:
+  - - "  return { success: false, message: \"Division by zero is not allowed\" };"
+  - - " return \"Division by zero is not allowed\" ;"
+- index: 1
+  absent: "return"
+  text: Use the return keyword to return the result of the division.
+  examples:
+  - - "  a / b ;"
+- index: 1
+  present: "{ (.*?)} "
+  text: Try simply returning the result of the division.
+  examples:
+  - - "  return { success: true, result: a / b };"
+  - - "  return { result: a / b };"
+- index: 2
+  absent: '\s*try\s*{\s* '
+  text: Use a try block to catch any exceptions that might be thrown.
+  examples:
+  - - "  const result = divide(10, 2);"
+- index: 2
+  present: '\s* try \s* { .*? if \( result.success \) .*?'
+  text: You may assume that the result is successful within the try block.
+  examples:
+  - - " try { const result = divide(10 ,2); if( result.success) { console.log ( \"Result:\", result ); "
+- index: 2
+  present: '.*? result.result .*?'
+  text: The result is not an object, it is a number.
+  examples:
+  - - " try { const result = divide(10 ,2); console.log ( \"Result:\", result.result ); "
+- index: 2
+  absent: '.*? catch .*? '
+  text: Handle the error within the catch block.
+  examples:
+  - - " try { const result = divide(10 ,2); console.log ( \"Result:\", result ); }"
+- index: 2
+  absent: '\s* catch \s* \( .*? \) { \s* '
+  text: Catch an error object within the catch block.
+  examples:
+  - - " try { const result = divide(10 ,2); console.log ( \"Result:\", result ); } catch {}"
+- index: 2
+  absent: '\s* console.error \(\s* "Error:" \s* , .*?.message \) '
+  text: The error is an object with a message property.
+  examples:
+  - - " try { const result = divide(10 ,2); console.log ( \"Result:\", result ); } catch (error) { console.error( \"Error:\", result);  } "
+# debug: true
+</script>
+</head>
+<body>
+<!-- For GitHub Pages formatting: -->
+<div class="container-lg px-3 my-5 markdown-body">
+<h1>Lab Exercise Handling Errors</h1>
+<p>
+This is a lab exercise on developing secure software.
+For more information, see the <a href="introduction.html" target="_blank">introduction to
+the labs</a>.
+
+<p>
+<h2>Task</h2>
+<p>
+<b>Please change the code below to leverage the exception handling mechanism and improve its robustness.</b>
+
+<p>
+<h2>Background</h2>
+<p>
+In this exercise, we'll demonstrate how to enhance the robustness of a JavaScript function by transitioning from a return code mechanism to exception handling. The primary goal is to understand how to handle errors more effectively, improving the readability and maintainability of the code.
+</p>
+Initially, the code uses a return code mechanism to handle potential errors. 
+The <tt>divide</tt> function returns an object indicating whether the operation was successful or if an error occurred. The caller must then check the success property and handle the result or error accordingly. This method works but can clutter the code with repetitive error-checking logic.
```

Comment:
Add:

"What's worse, it's easy for callers to ignore the error return code, which can result in surprising failures and security vulnerabilities".

We want to make sure they understand *why* they might make this change. It's also helpful to contrast this with languages like Rust & Haskell, which have a mechanism that at first *looks* like an error return, but *requires* thaat the caller deal with the error case.